doc/sources/installation/installation-rpi.rst

```diff
@@ -6,50 +6,134 @@ Installation on Raspberry Pi
 Raspberry Pi. Both options are described below.
 
+Raspberry Pi 4 manual installation (On Raspbian Buster)
+------------------------------------------------
+
+#. If you have installed Raspbian with a desktop, then you can skip to `Kivy dependencies`_.
+
+   If you are using a the Raspbian Lite, then you first need to install X::
+
+    sudo apt-get install xserver-xorg
+
+#. Next install nodm from source, so it includes the following fix: `spanezz/nodm#1 <https://github.com/spanezz/nodm/pull/10>`_::
+
+    sudo apt-get install libpam0g-dev help2man libx11-dev debhelper
+    git clone https://github.com/slashblog/nodm.git
+    pushd nodm
+        git checkout d48a8f6266d3f464138e0e95b65896917c35c89f
+        wget http://deb.debian.org/debian/pool/main/n/nodm/nodm_0.13-5.debian.tar.xz
+        tar xf nodm_0.13-5.debian.tar.xz
+        sudo dpkg-buildpackage -us -uc -b
+    popd
+    sudo dpkg -i nodm_0.13-*_armhf.deb
+    sudo rm -rf nodm*
+
+#. Now enable graphical login::
+
+    sudo systemctl set-default graphical.target
+
+#. Create a small test app::
+
+    cat <<EOF > ~/app.py
+    from kivy.app import App
+    from kivy.uix.button import Button
+
+
+    class TestApp(App):
+
+        def build(self):
+            return Button(text='hello world')
+
+
+    if __name__ == '__main__':
+        TestApp().run()
+    EOF
 
-Manual installation (On Raspbian Jessie/Stretch)
+#. It is recommend to use the the logger functionality instead of print statements: `<https://kivy.org/doc/stable/api-kivy.logger.html>`_, so they show up in the log.
+
+   A small script can then be used to follow the latest log and restart the application by using the ``--restart`` argument::
+
+    #!/bin/bash -e
+
+    if [[ $* == *--restart* ]]; then
+        sudo service nodm restart
+        inotifywait -q ~/.kivy/logs -e create --format %w%f | xargs tail -f
+    else
+        ls -t -d ~/.kivy/logs/* | head -n1 | xargs tail -f
+    fi
+
+   Note that you need to install `inotify-tools` in order to use the restart functionality.
+
+#. Configure nodm and start your app at boot::
+
+    # Has the same effect as calling 'sudo dpkg-reconfigure nodm'
+    sudo sh -c 'echo "NODM_ENABLED=true" > /etc/default/nodm'
+    sudo sh -c 'echo "NODM_USER=$SUDO_USER" >> /etc/default/nodm' # Note that the variable SUDO_USER is used
+    sudo sh -c 'echo "NODM_FIRST_VT='\''7'\''" >> /etc/default/nodm'
+    sudo sh -c 'echo "NODM_XSESSION=/etc/X11/Xsession" >> /etc/default/nodm'
+    sudo sh -c 'echo "NODM_X_OPTIONS='\''-nolisten tcp'\''" >> /etc/default/nodm'
+    sudo sh -c 'echo "NODM_MIN_SESSION_TIME=60" >> /etc/default/nodm'
+    sudo sh -c 'echo "NODM_X_TIMEOUT=300" >> /etc/default/nodm'
+
+    # Start the app using nodm
+    echo '#!/bin/bash' > ~/.xsession
+    echo 'export DISPLAY=:0.0' >> ~/.xsession
+    echo "python3 ~/app.py" >> ~/.xsession
+
+#. For installing the _`Kivy dependencies` refer to step 1-2 in the `Raspberry Pi 1-3 manual installation`_ instructions below.
```

Comment:
Maybe break here and start a new sub section called "Installing Kivy"or something? Then this section can be named installing Kivy and the previous sub section can be named something like "Installing kivy headless prerequisites" and it flow better together rather than being one long instruction list.

Reply:
Or maybe this whole section should just apply on the headless install and the next section, `Raspberry Pi 1-3 manual installation` should also include Pi4, but instead of the instructions to install from master you say that for Pi4 you can use a wheel?